### DISCUSSION TITLE
Is there a way to regenerate tickets PDF?

### DISCUSSION THREAD

**Discussion by HansF** [pretix/pretix#2060] [Q&A] [answered]
For some reason the data folder is empty, I think I fudged up somewhere. 

Is there a way to regenerate all the ticket pdfs?

**Comment by raphaelm** [comment, marked as answer]
Don't worry, pretix will regenerate each PDF as soon as someone tries to download it!

**Reply by HansF** [reply]
Thanks, @raphaelm I need to investigate more... 

![image](https://user-images.githubusercontent.com/1503/117137300-c91aff00-ada9-11eb-947a-63fa626264b1.png)

and 

![image](https://user-images.githubusercontent.com/1503/117137452-02ec0580-adaa-11eb-8e06-9cf86141c77c.png)

Permissions seems fine... 

![image](https://user-images.githubusercontent.com/1503/117137475-0ed7c780-adaa-11eb-961f-7e232f5cef5f.png)

Have to do something else now, will investigate later this evening.

**Reply by HansF** [reply]
ERROR 2021-05-05 11:22:32,133 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:22:32,134 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:22:58,320 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:22:58,321 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:23:07,885 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:23:07,886 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:24:24,365 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:24:24,365 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:29:16,446 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:29:16,446 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:31:06,561 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:31:06,562 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:41:51,488 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:41:51,491 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:41:56,853 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:41:56,855 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
INFO 2021-05-05 11:45:07,638 celery.app.trace trace Task pretix.base.services.quotas.refresh_quota_caches[1f1e97dc-7b40-4b4f-a973-ab654219bb0e] succeeded in 0.01485113799571991s: None
ERROR 2021-05-05 11:56:49,578 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
ERROR 2021-05-05 11:56:49,580 pretix.base.tasks tasks Unexpected exception: FileNotFoundError(2, 'No such file or directory')
root@pretixvirgininstall-s-1vcpu-2gb-ams3-01:/var/pretix/data/media/tickets/RR/RRC21#

**Reply by raphaelm** [reply]
Ah, so not the ticket PDF itself is missing, but the background picture you used! Uhm, I don't think there's an official way out. You could try manually removing the links to the files from the database (``ticketoutputpdf_ticketlayout`` or a similar table), then clearing the cache and manually uploading the files again.

Do you have a backup of the media dir?

**Reply by HansF** [reply]
Thanks @raphaelm, great reply. 
I've restored the media/pub folder from a backup and now we're back in business.
